File: scripts/trust_codex_project.js
#!/usr/bin/env node
"use strict";

const fs = require("node:fs");
const os = require("node:os");
const path = require("node:path");

const TOML_PATH = path.join(os.homedir(), ".codex", "config.toml");

function usage() {
  process.stderr.write("usage: trust_codex_project.js <path>\n");
}

function main(argv = process.argv.slice(2)) {
  if (argv.length !== 1) {
    usage();
    return 1;
  }

  const projectPath = fs.realpathSync(path.resolve(argv[0]));
  const section = `[projects."${projectPath}"]`;
  let content = "";
  if (fs.existsSync(TOML_PATH)) {
    content = fs.readFileSync(TOML_PATH, "utf-8");
  }

  if (content.includes(section)) {
    return 0;
  }

  const entry = `${section}\ntrust_level = "trusted"\n`;
  fs.mkdirSync(path.dirname(TOML_PATH), { recursive: true });
  if (content && !content.endsWith("\n")) {
    content += "\n";
  }
  fs.writeFileSync(TOML_PATH, `${content}\n${entry}`, "utf-8");
  return 0;
}

if (require.main === module) {
  process.exitCode = main();
}

module.exports = { main };

File: archive/legacy-python/scripts/trust_codex_project.py
"""Ensure a project path is listed as trusted in ~/.codex/config.toml.

Codex shows an interactive "Do you trust this directory?" prompt on the
first launch inside an unknown cwd. When codex is spawned via the xmux
bridge, that prompt's `› 1. Yes, continue` option line matches the
bridge's idle pattern `^[[:space:]]*›` and causes a false idle detection.
The bridge then chunks its queued message into the trust prompt, which
codex interprets as invalid input and terminates the pane — producing
the "chunk paste-buffer failed / pane is gone" failure observed in
fe-debug-pipeline.

Pre-populating the projects table with `trust_level = "trusted"` makes
codex skip the trust prompt, so the bridge reaches the real prompt and
its idle detection operates on the normal input line only.

Idempotent: re-trusting an already-trusted path is a no-op.

Usage:
  python3 trust_codex_project.py <path>
"""
import os
import sys

TOML_PATH = os.path.expanduser("~/.codex/config.toml")


def main():
    if len(sys.argv) != 2:
        print("usage: trust_codex_project.py <path>", file=sys.stderr)
        sys.exit(1)

    path = os.path.realpath(os.path.abspath(sys.argv[1]))
    section = f'[projects."{path}"]'

    try:
        with open(TOML_PATH) as f:
            content = f.read()
    except FileNotFoundError:
        content = ""

    if section in content:
        return  # already trusted

    entry = f'{section}\ntrust_level = "trusted"\n'
    os.makedirs(os.path.dirname(TOML_PATH), exist_ok=True)
    if content and not content.endswith("\n"):
        content += "\n"
    content = content + "\n" + entry
    with open(TOML_PATH, "w") as f:
        f.write(content)


if __name__ == "__main__":
    main()
